### PR TITLE
Add setBranchWeigths convenience function. NFC

### PR DESCRIPTION
**Pull request by MatzeB**
Add `setBranchWeights` convenience function to ProfDataUtils.h and use
it where appropriate.

### CODE REVIEW

File: llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp [hunk 1]
```diff
@@ -1437,12 +1437,11 @@ void PGOUseFunc::populateCoverage(IndexedInstrProfReader *PGOReader) {
     // If A is uncovered, set weight=1.
     // This setup will allow BFI to give nonzero profile counts to only covered
     // blocks.
-    SmallVector<unsigned, 4> Weights;
+    SmallVector<uint32_t, 4> Weights;
```
Comment (ilovepi) [inline llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp:1440]: good catch!

**Comment by MatzeB** [comment]
Adding a global `setBranchWeights` convience function as requested in review of https://reviews.llvm.org/D159322

**Review by mtrofin — APPROVED** [review]
nit: should the commit title say [nfc]?

**Review by ilovepi — APPROVED** [review]
LGTM. I was surprised that `Instruction` didn't implement this already, but I think this is a good change. Thanks for the nice improvement.

**Pull request merged**